openmc/tallies.py

```diff
@@ -0,0 +1,3204 @@
+        nuclide : openmc.Nuclide
+            Nuclide to remove
+
+        """
+
+        if nuclide not in self.nuclides:
+            msg = 'Unable to remove nuclide "{0}" from Tally ID="{1}" since the ' \
+                  'Tally does not contain this nuclide'.format(nuclide, self.id)
+            ValueError(msg)
+
+        self._nuclides.remove(nuclide)
+
+    def _can_merge_filters(self, other):
+        """Determine if another tally's filters can be merged with this one's
+
+        The types of filters between the two tallies must match identically.
+        The bins in all of the filters must match identically, or be mergeable
+        in only one filter. This is a helper method for the can_merge(...)
+        and merge(...) methods.
+
+        Parameters
+        ----------
+        other : openmc.Tally
+            Tally to check for mergeable filters
+
+        """
+
+        # Two tallies must have the same number of filters
+        if len(self.filters) != len(other.filters):
+            return False
+
+        # Return False if only one tally has a delayed group filter
+        tally1_dg = self.contains_filter(openmc.DelayedGroupFilter)
+        tally2_dg = other.contains_filter(openmc.DelayedGroupFilter)
+        if sum([tally1_dg, tally2_dg]) == 1:
+            return False
+
+        # Look to see if all filters are the same, or one or more can be merged
+        for filter1 in self.filters:
+            merge_filters = False
+            mergeable_filter = False
+
+            for filter2 in other.filters:
+
+                # If filters match, they are mergeable
+                if filter1 == filter2:
+                    mergeable_filter = True
+                    break
+
+                # If filters are first mergeable filters encountered
+                elif filter1.can_merge(filter2) and not merge_filters:
+                    merge_filters = True
+                    mergeable_filter = True
+                    break
+
+                # If filters are the second mergeable filters encountered
+                elif filter1.can_merge(filter2) and merge_filters:
+                    return False
+
+            # If no mergeable filter was found, the tallies are not mergeable
+            if not mergeable_filter:
+                return False
+
+        # Tally filters are mergeable if all conditional checks passed
+        return True
+
+    def _can_merge_nuclides(self, other):
+        """Determine if another tally's nuclides can be merged with this one's
+
+        The nuclides between the two tallies must be mutually exclusive or
+        identically matching. This is a helper method for the can_merge(...)
+        and merge(...) methods.
+
+        Parameters
+        ----------
+        other : openmc.Tally
+            Tally to check for mergeable nuclides
+
+        """
+
+        no_nuclides_match = True
+        all_nuclides_match = True
+
+        # Search for each of this tally's nuclides in the other tally
```

Comment:
an algorithmically faster way to do this would be to add all elements to a set. If the length of the set is `len(self.nuclides) + len(other.nuclides)` or `len(self.nuclides)`, then return true.